src/v/cluster/partition.cc

```diff
@@ -208,21 +208,26 @@ ss::future<std::error_code> partition::prefix_truncate(
       clusterlog.info,
       "Truncating {} to redpanda offset {} kafka offset {}",
       _raft->ntp(),
-      truncation_offset,
-      kafka_truncation_offset);
-    if (truncation_offset != model::offset{}) {
-        auto err = co_await _log_eviction_stm->truncate(
-          truncation_offset, deadline, _as);
-        if (err) {
-            co_return err;
-        }
+      rp_start_offset,
+      kafka_start_offset);
+    auto err = co_await _log_eviction_stm->truncate(
+      rp_start_offset, kafka_start_offset, deadline, _as);
+    if (err) {
+        co_return err;
     }
-    // TODO(awong): have archival listen on log_eviction_stm.
     if (_archival_meta_stm) {
-        co_return co_await _archival_meta_stm->truncate(
-          kafka_truncation_offset, deadline, _as);
+        // The archival metadata stm also listens for prefix_truncate batches.
+        auto applied = co_await _archival_meta_stm->wait_no_throw(
+          _raft->committed_offset(), deadline, _as);
```

Comment:
Why wait on raft committed? I would assume `log_eviction_stm->truncate` to return the offset that `replicate` returned, and that can be used as the wait offset, is it because log_eviction_stm->truncate also performs the wait?

Reply:
Yeah good point. This was mostly because I didn't feel like doing the plumbing, but looking at it, it shouldn't be too bad.

May do this in a follow-up.